### PR TITLE
im: select n-th candidate when number n is pressed

### PR DESCRIPTION
**Pull request by johnchen902**
This is in the todo list (#1), and I agree that it's quite useful.

My keyboard does not have a keypad, so you may need to test that part yourself.

### CODE REVIEW

File: im.c [hunk 1]
```diff
@@ -17,6 +17,17 @@ static void noop() {
 
 static void vte_hack(struct wlchewing_state *state);
 
+static bool commit_bottom_panel(struct wlchewing_state *state, int offset) {
+	int index = state->bottom_panel->selected_index + offset;
+	if (index >= chewing_cand_TotalChoice(state->chewing))
+		return false;
```
Comment (xdavidwu) [inline im.c:23]: style: we always use brackets

File: im.c [hunk 2]
```diff
@@ -68,12 +79,15 @@ int im_key_press(struct wlchewing_state *state, uint32_t key) {
...
-			bottom_panel_destroy(state->bottom_panel);
-			state->bottom_panel = NULL;
-			need_update = true;
+			need_update = commit_bottom_panel(state, 0);
+			break;
+		case XKB_KEY_1 ... XKB_KEY_9:
+			need_update = commit_bottom_panel(state,
+					keysym - XKB_KEY_1);
+			break;
+		case XKB_KEY_KP_1 ... XKB_KEY_KP_9:
```
Comment (xdavidwu) [inline im.c:88]: I think we should map 0 to 10th candidate too.

**Review by xdavidwu — CHANGES_REQUESTED** [review]

**Comment by johnchen902** [comment]
Made the requested changes. PTAL

**Review by xdavidwu — APPROVED** [review]
LGTM

**Comment by xdavidwu** [comment]
Thanks!

**Pull request merged**